src/autocomplete/dataset.js

```diff
@@ -93,7 +93,7 @@ _.mixin(Dataset.prototype, EventEmitter, {
     var renderArgs = [].slice.call(arguments, 2);
     this.$el.empty();
 
-    hasSuggestions = suggestions && suggestions.length;
+    hasSuggestions = suggestions && suggestions.length > 0;
```

Comment:
Those tests are essentially the same when dealing with arrays, that's why I am trying to dig this issue to understand it.